src/main/java/soonkeatneo/duke/TaskList.java

```diff
@@ -0,0 +1,186 @@
+package soonkeatneo.duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import soonkeatneo.duke.task.Task;
+import soonkeatneo.duke.task.Deadline;
+import soonkeatneo.duke.task.Event;
+import soonkeatneo.duke.task.Todo;
+
+/**
+ * Stores and manages the list of {@Task} created.
+ * @author Soon Keat Neo
+ * @version CS2103T AY20/21 Sem 1 iP v0.1
+ */
+public class TaskList {
+    private static List<Task> taskList;
+
+    /**
+     * Creates a new task list without reading from existing storage.
+     */
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Overloaded constructor creating a new task list reading from existing {@Storage}.
+     * @param storage Storage object to be read from
+     */
+    public TaskList(Storage storage) {
+        this.taskList = new ArrayList<>();
+        initializeTasks(storage.read());
+    }
+
+    /**
+     * Prints a list of {@Task} matching the given string.
+     * @param inputString the String to search for
+     */
+    public void find(String inputString) {
+        List<Task> newList = new ArrayList<>();
+        boolean hasPrintedTasks = false;
+        int counter = 0;
+        for (int i = 0; i < this.taskList.size(); i++) {
+            Task task = this.taskList.get(i);
+            if (task.getDescription().toLowerCase().contains(inputString.toLowerCase())) {
+                if (!hasPrintedTasks) {
+                    Ui.printMessage("Here are the list of matching tasks: ");
+                    hasPrintedTasks = true;
+                }
+                Ui.printMessage((counter + 1) + ". " + task.toString());
+                counter += 1;
+            }
+        }
+        if (!hasPrintedTasks) {
+            Ui.printMessage("There are no matching tasks!");
+        }
+    }
+
+    /**
+     * Adds a new {@Task} to the task list.
+     * @param task Given task to be added to the task list
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+        Ui.printMessage("Wakarimashita! Task added to list:");
+        Ui.printMessage(task.toString());
+        Ui.printMessage("The size of your task list is now: " + this.getSize());
+    }
+
+    /**
+     * Mark the specified task as complete.
+     * @param inputString User input string
+     */
+    public void completeTask(String inputString, Storage storage) {
+        try {
+            int taskId = Integer.parseInt(String.valueOf(inputString.split(" ")[1])) - 1;
```

Comment:
The () becomes a bit deep here? I am _not sure_ whether this is related to code quality but I guess can extract out a few lines out of this single line to make it more readable? 🤔